server/api/post_action.go

```diff
@@ -70,14 +71,47 @@ func (api *api) postActionTentative(w http.ResponseWriter, req *http.Request) {
 }
 
 func (api *api) postActionRespond(w http.ResponseWriter, req *http.Request) {
-	mscalendar, user, eventID, option := api.preprocessAction(w, req)
+	calendar, user, eventID, option, postID := api.preprocessAction(w, req)
 	if eventID == "" {
 		return
 	}
-	err := mscalendar.RespondToEvent(user, eventID, option)
-	if err != nil {
-		// h.LogWarn(err.Error())
+	err := calendar.RespondToEvent(user, eventID, option)
+	if err != nil && !strings.HasPrefix(err.Error(), "202") {
 		http.Error(w, "Failed to respond to event: "+err.Error(), http.StatusBadRequest)
 		return
 	}
+
+	var response string
+	switch option {
+	case mscalendar.OptionYes:
+		response = mscalendar.ResponseYes
+	case mscalendar.OptionNo:
+		response = mscalendar.ResponseNo
+	case mscalendar.OptionMaybe:
+		response = mscalendar.ResponseMaybe
+	default:
+		return
+	}
+
+	p, err := api.PluginAPI.GetPost(postID)
+	if err != nil {
+		http.Error(w, "Failed to update the post: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	sas := p.Attachments()
+	if len(sas) == 0 {
+		http.Error(w, "Failed to update the post: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	sa := sas[0]
+	sa.Actions = mscalendar.GetPostActionSelect(eventID, response, req.URL.String())
```

Comment:
It doesn't seem obvious to me that `GetPostActionSelect` is specific to responding to an event. Maybe rename to `NewPostActionForEventResponse`